### PR TITLE
feat: add key did provider secp256k1 package

### PR DESCRIPTION
**Pull request by zachferland**
Moves https://github.com/ceramicnetwork/key-did-provider-secp256k1, but that package was a bit out of date, this mostly copies over everything from `key-did-provider-ed25519` for `secp256k1` instead

### CODE REVIEW

**Review by oed — APPROVED** [review]
lgtm, Are we making a new release after merge?

**Comment by zachferland** [comment]
yeah we can release, I dont have permissions for this package if you wan to add them, cc @oed

**Pull request merged**
